### PR TITLE
doc: Replace a link to Qt precompiled binaries with compile instructions

### PR DESCRIPTION
**Pull request by hebasto**
This PR replaces a [link to Qt precompiled binaries](https://github.com/sipsorcery/qt_win_binary/releases) with compile instructions that allow users to self-compile static Qt package which is required for building Bitcoin Core with Visual Studio.

### CODE REVIEW

**Comment by hebasto** [comment]
cc @sipsorcery @Bosch-0 @jarolrod

**Comment by sipsorcery** [comment]
@hebasto concept ACK on the doc changes.

Changing the Qt path to `C:\Qt_static` has the potential to make it trickier to switch between Qt versions, however, I suspect that may be a problem specific to me.

**Comment by hebasto** [comment]
Updated 7b8b6c80384f99c7ca864c821e1b967a4505c185 -> ecb823e4b0eb2bfab43fc964278a18ebcdf46b5e ([pr22890.01](https://github.com/hebasto/bitcoin/commits/pr22890.01) -> [pr22890.02](https://github.com/hebasto/bitcoin/commits/pr22890.02), [diff](https://github.com/hebasto/bitcoin/compare/pr22890.01..pr22890.02)):

- addressed @sipsorcery's [comment](https://github.com/bitcoin/bitcoin/pull/22890#issuecomment-913110148):

> Changing the Qt path to `C:\Qt_static` has the potential to make it trickier to switch between Qt versions, however, I suspect that may be a problem specific to me.

Now a non-default path to static Qt could specified in the `QTBASEDIR` environment variable.

**Comment by hebasto** [comment]
Updated ecb823e4b0eb2bfab43fc964278a18ebcdf46b5e -> 42df8d519d20897dabb3ca07eec6769304d1ff37 ([pr22890.02](https://github.com/hebasto/bitcoin/commits/pr22890.02) -> [pr22890.03](https://github.com/hebasto/bitcoin/commits/pr22890.03), [diff](https://github.com/hebasto/bitcoin/compare/pr22890.02..pr22890.03)):

- added commit which drops manual Qt source code patching, see: https://github.com/qt/qtbase/commit/fdb7fa937a58900e3966adc765975785d0a9f0da.

**Comment by hebasto** [comment]
Updated 42df8d519d20897dabb3ca07eec6769304d1ff37 -> 8e9410f188e1fa12cdd5b5b5bae4df8445cd385e ([pr22890.03](https://github.com/hebasto/bitcoin/commits/pr22890.03) -> [pr22890.04](https://github.com/hebasto/bitcoin/commits/pr22890.04), [diff](https://github.com/hebasto/bitcoin/compare/pr22890.03..pr22890.04)):

- dropped `-ltcg` configure option because [link-time code generation](https://docs.microsoft.com/en-us/cpp/build/reference/ltcg-link-time-code-generation?view=msvc-160) significantly increase (at times) the required disc space, and it should not be suggested to users by default.

File: build_msvc/README.md [hunk 1]
```diff
@@ -3,67 +3,72 @@ Building Bitcoin Core with Visual Studio
 
 Introduction
 ---------------------
-Solution and project files to build the Bitcoin Core applications `msbuild` or Visual Studio can be found in the `build_msvc` directory. The build has been tested with Visual Studio 2019 (building with earlier versions of Visual Studio should not be expected to work).
+Solution and project files to build the Bitcoin Core with applications `msbuild` or Visual Studio can be found in the `build_msvc` directory. The build has been tested with Visual Studio 2019 (building with earlier versions of Visual Studio should not be expected to work).
```
Comment (jarolrod) [inline build_msvc/README.md:6]: while here, some slight cleanups would be nice.
```suggestion
Solution and project files to build Bitcoin Core with `msbuild` or Visual Studio can be found in the `build_msvc` directory. The build has been tested with Visual Studio 2019 (building with earlier versions of Visual Studio should not be expected to work).
```

File: build_msvc/README.md [hunk 2]
```diff
@@ -3,67 +3,72 @@ Building Bitcoin Core with Visual Studio
...
-vcpkg integrate install
+Add-Content -Path "vcpkg\triplets\x64-windows-static.cmake" -Value "set(VCPKG_BUILD_TYPE release)"
 ```
 
 Qt
 ---------------------
-In order to build Bitcoin Core a static build of Qt is required. The runtime library version (e.g. v142) and platform type (x86 or x64) must also match.
+To build Bitcoin Core with the GUI, a static build of Qt is required.
+
+1. Download a single ZIP archive of Qt source from https://download.qt.io/official_releases/qt/ (e.g., [`qt-everywhere-src-5.12.11.zip`](https://download.qt.io/official_releases/qt/5.12/5.12.11/single/qt-everywhere-src-5.12.11.zip)), and expand it into a dedicated folder. In this document it is assumed that this folder is `C:\dev\qt-source`.
```
Comment (jarolrod) [inline build_msvc/README.md:38]: "Qt source" -> "Qt source code" is just a nit
```suggestion
1. Download a single ZIP archive of Qt source code from https://download.qt.io/official_releases/qt/ (e.g., [`qt-everywhere-src-5.12.11.zip`](https://download.qt.io/official_releases/qt/5.12/5.12.11/single/qt-everywhere-src-5.12.11.zip)), and expand it into a dedicated folder. The following instructions assume that this folder is `C:\dev\qt-source`.
```

**Review by jarolrod — COMMENTED** [review]
Some notes on wording, I will test for correctness of instructions later.

*Prepares hazmat suit to enter the windows environment*

**Comment by hebasto** [comment]
Updated 8e9410f188e1fa12cdd5b5b5bae4df8445cd385e -> c948757c99c5e1c80d4edc1ab201ef93b7803d70 ([pr22890.04](https://github.com/hebasto/bitcoin/commits/pr22890.04) -> [pr22890.05](https://github.com/hebasto/bitcoin/commits/pr22890.05), [diff](https://github.com/hebasto/bitcoin/compare/pr22890.04..pr22890.05)):

- addressed @jarolrod's comments

File: build_msvc/README.md [hunk 3]
```diff
@@ -3,67 +3,72 @@ Building Bitcoin Core with Visual Studio
 
 Introduction
 ---------------------
-Solution and project files to build the Bitcoin Core applications `msbuild` or Visual Studio can be found in the `build_msvc` directory. The build has been tested with Visual Studio 2019 (building with earlier versions of Visual Studio should not be expected to work).
+Solution and project files to build the Bitcoin Core with applications `msbuild` or Visual Studio can be found in the `build_msvc` directory. The build has been tested with Visual Studio 2019 (building with earlier versions of Visual Studio should not be expected to work).
 
-Building with Visual Studio is an alternative to the Linux based [cross-compiler build](https://github.com/bitcoin/bitcoin/blob/master/doc/build-windows.md).
+To build Bitcoin Core from a command-line interface, it is sufficient to install the Build Tools component only.
```
[the remark this answers is not in the document]
Comment (hebasto) [reply build_msvc/README.md]: Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/22890#issuecomment-913193222).

File: build_msvc/README.md (hunk 1 above)
Comment (hebasto) [reply build_msvc/README.md]: Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/22890#issuecomment-913193222).

File: build_msvc/README.md (hunk 2 above)
Comment (hebasto) [reply build_msvc/README.md]: Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/22890#issuecomment-913193222).

File: build_msvc/README.md [hunk 4]
```diff
@@ -3,67 +3,72 @@ Building Bitcoin Core with Visual Studio
...
 
 ```
 PS >py -3 msvc-autogen.py
 ```
 
-- An optional step is to adjust the settings in the `build_msvc` directory and the `common.init.vcxproj` file. This project file contains settings that are common to all projects such as the runtime library version and target Windows SDK version. The Qt directories can also be set.
+2. An optional step is to adjust the settings in the `build_msvc` directory and the `common.init.vcxproj` file. This project file contains settings that are common to all projects such as the runtime library version and target Windows SDK version. The Qt directories can also be set. To specify a non-default path to a static Qt package directory, use the `QTBASEDIR` environment variable.
 
-- To build from the command line with the Visual Studio 2019 toolchain use:
+3. To build from the command line with the Visual Studio 2019 toolchain use:
```
Comment (jarolrod) [inline build_msvc/README.md:65]: sorry, another nit (for consistency). Apply, only when you need to retouch for a good reason 
```suggestion
3. To build from the command-line with the Visual Studio 2019 toolchain use:
```

**Comment by fanquake** [comment]
There's no point adding the patching instructions in 9a434e81f8dd5c1b97442ba1cbb521856d8375b3, just to drop them in a later commit (c948757c99c5e1c80d4edc1ab201ef93b7803d70).

**Comment by hebasto** [comment]
Updated c948757c99c5e1c80d4edc1ab201ef93b7803d70 -> 0f56ce36483f01da8ed2b6b35a3484649cfe5d7d ([pr22890.05](https://github.com/hebasto/bitcoin/commits/pr22890.05) -> [pr22890.06](https://github.com/hebasto/bitcoin/commits/pr22890.06), [diff](https://github.com/hebasto/bitcoin/compare/pr22890.05..pr22890.06)):

- addressed @jarolrod's [comment](https://github.com/bitcoin/bitcoin/pull/22890#discussion_r702476351)
- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/22890#issuecomment-913283332)

File: build_msvc/README.md [hunk 5]
```diff
@@ -3,67 +3,72 @@ Building Bitcoin Core with Visual Studio
...
-msbuild /m bitcoin.sln /p:Platform=x64 /p:Configuration=Release /t:build
+```cmd
+msbuild -property:Configuration=Release -maxCpuCount -verbosity:minimal bitcoin.sln
 ```
 
-- Alternatively, open the `build_msvc/bitcoin.sln` file in Visual Studio 2019.
+Alternatively, open the `build_msvc/bitcoin.sln` file in Visual Studio 2019.
 
 AppVeyor
 ---------------------
```
Comment (sipsorcery) [inline build_msvc/README.md:74]: I guess the AppVeyor section will be removed when #21551 gets merged?

**Comment by hebasto** [comment]
Rebased 0f56ce36483f01da8ed2b6b35a3484649cfe5d7d -> 391a377fc9a7261a2f73221150c599adf7076fa9 ([pr22890.06](https://github.com/hebasto/bitcoin/commits/pr22890.06) -> [pr22890.07](https://github.com/hebasto/bitcoin/commits/pr22890.07)) due to the conflict with #21551.

File: build_msvc/README.md (hunk 5 above)
Comment (hebasto) [reply build_msvc/README.md]: [Rebased](https://github.com/bitcoin/bitcoin/pull/22890#issuecomment-914150280) on top of #21551.

**Comment by hebasto** [comment]
Rebased 391a377fc9a7261a2f73221150c599adf7076fa9 -> 367203bc93a9016519221e4b88fc2b1b13d91279 ([pr22890.07](https://github.com/hebasto/bitcoin/commits/pr22890.07) -> [pr22890.08](https://github.com/hebasto/bitcoin/commits/pr22890.08)) due to the conflict with #22899.

**Comment by laanwj** [comment]
> that allow users to self-compile static Qt package which is required for building Bitcoin Core with Visual Studio.

Why is a static Qt package necessarily needed? I mean, on Linux it works fine to build it against a dynamic system Qt.

**Comment by hebasto** [comment]
@laanwj 

> > that allow users to self-compile static Qt package which is required for building Bitcoin Core with Visual Studio.
> 
> Why is a static Qt package necessarily needed? I mean, on Linux it works fine to build it against a dynamic system Qt.

#### A relevant discussion from #15529:

@NicolasDorier https://github.com/bitcoin/bitcoin/pull/15529#issuecomment-507160234:
> Btw, would it be a good idea to use QT's precompiled dynamic library instead of the static lib?
We are using the msvc build mainly for debugging stuff so I don't think it would matter too much and prevent a bunch of headache.

@sipsorcery https://github.com/bitcoin/bitcoin/pull/15529#issuecomment-507163992
> @NicolasDorier nooooo :(. You don't want to know how long I've spent getting the Qt build working.
>
> Putting aside the lost chunk of my life it would be a big change switching the build to dynamic linking. All the Bitcoin Core libraries would have to be built as dll's and there are then bound to be missing exports that will need to be added which would mean code changes.
>
> I'd highly recommend sticking with static linking for consistency with the other Bitcoin Core builds. Hopefully one day the `vcpkg` static build of Qt will work and allow it to be treated the same as the other dependencies for the msvc build. See microsoft/vcpkg#4560 (comment).

**Comment by sipsorcery** [comment]
> Why is a static Qt package necessarily needed? I mean, on Linux it works fine to build it against a dynamic system Qt.

To further elaborate, to the best of my knowledge a static version of Qt is required for the Bitcoin Core version on Windows because we want the `bitcoin-qt.exe` and other executables to be statically linked against the C runtime (the [MT compiler option](https://docs.microsoft.com/en-us/cpp/build/reference/md-mt-ld-use-run-time-library?view=msvc-160)).

The `msvc` compiler options page also has this quote:

> All modules passed to a given invocation of the linker must have been compiled with the same run-time library compiler option (/MD, /MT, /LD).

I was never able to build `bitcoim-qt.exe` with a version of Qt compiled with the `MD` compiler option but perhaps I'm missing something.

**Comment by laanwj** [comment]
OK, thanks for explaining, I was under the impression it was something new introduced here, but if it's simply the way things have always been…"it's a lot of extra work" is enough reason to not do it imo. FWIW, on Linux it's the othe way around, dynamically building is "go with the flow" whereas static is what requires a lot of extra work.
 
Concept ACK

File: build_msvc/README.md [hunk 6]
```diff
@@ -3,67 +3,72 @@ Building Bitcoin Core with Visual Studio
...
 
-Dependencies
+Prerequisites
 ---------------------
-A number of [open source libraries](https://github.com/bitcoin/bitcoin/blob/master/doc/dependencies.md) are required in order to be able to build Bitcoin Core.
+To build [dependencies](../doc/dependencies.md) (except for [Qt](#qt)),
+the default approach is to use the [vcpkg](https://docs.microsoft.com/en-us/cpp/vcpkg) package manager from Microsoft.
 
-Options for installing the dependencies in a Visual Studio compatible manner are:
+To install vcpkg:
```
Comment (fanquake) [inline build_msvc/README.md:18]: Please don't add instructions for installing a package manager here. It's got nothing todo with building Bitcoin Core, and is just another thing that can become outdated. Linking to upstream (https://github.com/Microsoft/vcpkg), where the instructions will always be correct, should be sufficient, and is what was being done previously.

**Comment by hebasto** [comment]
Updated 367203bc93a9016519221e4b88fc2b1b13d91279 -> 6bc43989372386bf1dc6ee2a763a46a2d0780059 ([pr22890.08](https://github.com/hebasto/bitcoin/commits/pr22890.08) -> [pr22890.09](https://github.com/hebasto/bitcoin/commits/pr22890.09), [diff](https://github.com/hebasto/bitcoin/compare/pr22890.08..pr22890.09)):

- addressed @fanquake's [comment](https://github.com/bitcoin/bitcoin/pull/22890#discussion_r719089817)

File: build_msvc/README.md (hunk 6 above)
Comment (hebasto) [reply build_msvc/README.md]: Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/22890#issuecomment-930993678).

**Pull request merged**